gprofiler/spark/sampler.py

```diff
@@ -47,204 +39,19 @@ def __init__(
-            return True
-        elif rm1_address.startswith(host_name):
-            logger.info(
-                f"This is the collector master, because rm1: {rm1_address!r}"
-                f" starts with my host name: {host_name!r}, enabling Spark collector"
-            )
-            return True
-        else:
-            logger.info(
-                f"This is not the collector master, because rm1: {rm1_address!r}"
-                f" does not start with my host name: {host_name!r}, skipping Spark collector on this YARN master"
-            )
-            return False
-
-    def _get_spark_manager_process(self) -> Optional[psutil.Process]:
-        def is_master_process(process: psutil.Process) -> bool:
-            try:
-                return (
-                    "org.apache.hadoop.yarn.server.resourcemanager.ResourceManager" in process.cmdline()
-                    or "org.apache.spark.deploy.master.Master" in process.cmdline()
-                    or "mesos-master" in process_exe(process)
-                )
-            except MissingExePath:
-                return False
-
-        try:
-            return next(search_for_process(is_master_process))
-        except StopIteration:
-            return None
-
-    def _find_spark_cluster(self) -> Optional[Tuple[str, str]]:
-        """:return: (master address, cluster mode)"""
-        spark_master_process = self._get_spark_manager_process()
-        spark_cluster_mode = "unknown"
-        webapp_url = None
-
-        if spark_master_process is None:
-            logger.debug("Could not find any spark master process (resource manager or spark master)")
-            return None
-
-        if "org.apache.hadoop.yarn.server.resourcemanager.ResourceManager" in spark_master_process.cmdline():
-            if not self._is_yarn_master_collector(spark_master_process):
-                return None
-            spark_cluster_mode = SPARK_YARN_MODE
-            webapp_url = self._guess_yarn_resource_manager_webapp_address(spark_master_process)
-        elif "org.apache.spark.deploy.master.Master" in spark_master_process.cmdline():
-            spark_cluster_mode = SPARK_STANDALONE_MODE
-            webapp_url = self._guess_standalone_master_webapp_address(spark_master_process)
-        elif "mesos-master" in process_exe(spark_master_process):
-            spark_cluster_mode = SPARK_MESOS_MODE
-            webapp_url = self._guess_mesos_master_webapp_address(spark_master_process)
-
-        if spark_master_process is None or webapp_url is None or spark_cluster_mode == "unknown":
-            logger.warning("Could not get proper Spark cluster configuration")
-            return None
-
-        logger.info("Guessed settings are", cluster_mode=spark_cluster_mode, webapp_url=webapp_url)
-
-        return webapp_url, spark_cluster_mode
+        self._spark_sampler = BigDataSampler(
+            logger, get_hostname(), self._master_address, self._spark_mode, self._applications_metrics
```

Comment:
As `master_address` is not configurable - just pass None?

Reply:
Same for `self._spark_mdoe` and `self._applications_metrics` - Just remove them and pass None/False, explaining that the autofill logic in BIgDataSampler will handle?

Reply:
IDC to pass `None`
btw TODO is to have the option to configure them in the future

Reply:
Mhm, you can leave such a TODO in code.

Reply:
https://github.com/Granulate/gprofiler/pull/749/files#diff-18dce65a9628cdb97bacc94da9b0401f8686d3326c5a71c91a23eb0bf4cbb919R49-R52